security/Kconfig

```diff
@@ -42,6 +42,28 @@ config SECURITY_TIOCSTI_RESTRICT
 
 	  If you are unsure how to answer this question, answer N.
 
+config SECURITY_MODHARDEN
+	bool "Harden module auto-loading"
+	default y
```

Comment:
I think for now we should default this to n as it has quite severe implications.
Maybe it would be worth to be able to use the cmdline to override the value on boot

Reply:
> Maybe it would be worth to be able to use the cmdline to override the value on boot

I don't think there'd be much more advantage than a sysctl.

Reply:
The difference is that changing auto module loading behavior can be quite intrusive and potentially even lock out user sessions depending on the environment. automatic and convenient global sysctl configs may or may not be available depending on the userland. Therefor I believe its a good idea to have a kernel cmdline option to override the default value

Reply:
Ok, that makes sense.

So it should be something like this just below https://github.com/madaidan/linux-hardened/blob/modharden/kernel/kmod.c#L109?

```
static int __init enable_modharden(char *level)
{
	if (!level)
		return -EINVAL;

	if (strcmp(level, "on") == 0)
		modharden = 1;
	else if (strcmp(level, "off") == 0)
		modharden = 0;
	else
		return -EINVAL;

	return 0;
}
early_param("modharden", enable_modharden);
```

Reply:
Did that https://github.com/madaidan/linux-hardened/commit/414cf566963dbc759543b044764892dcc93d3b51